sample/src/main/java/dev/marcellogalhardo/retained/sample/SampleView.kt

```diff
@@ -0,0 +1,21 @@
+package dev.marcellogalhardo.retained.sample
+
+import android.content.Context
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import dev.marcellogalhardo.retained.view.retain
+
+class SampleView(context: Context, attrs: AttributeSet) : View(context, attrs) {
+
+    private val presenter: SimplePresenter by retain {
+        SimplePresenter()
```

Comment:
I like how this looks like for this sample, but I have concerns when dealing with complex objects.

For example, how can people inject dependencies here? One option is to use `inject,` but it might come with many problems when using with views. Another option I see is to use `application.getSystemService<PresenterProvider>()` and use it as an "a la carta" service locator. Still, developers would need to be extra careful as they might break previews, etc (the dependencies can not be provided during preview).

Reply:
Yeah this is a valid point. We could either argue it's out of the scope of the library and be done with it, since there are many different solutions people can use, or we could use this as an argument against adding the view implementation in the library, since this may push people to try to deal with this awkward problem.

>One option is to use .inject(target: View), but it might come with many problems when using with views.

Do you mind giving some more details around these problems?

And just to add one more option to the solutions here, we can always expose the dependency we need in a Dagger component and do something like this:

```kotlin
inline val View.injector
  get() = (findFragment<Fragment>().requireActivity().application as YourComponentProvider).yourComponent
// something slightly different for cases without fragments
```

And then:

```kotlin
private val presenter by retain { injector.simplePresenter }
```

Doesn't look great, but that's probably how I'd do it.

Reply:
Actually forget about that Dagger example, it's easier to use Hilt as a better solution since it actually has proper support for views!